lib/middleware/auth.js

```diff
@@ -0,0 +1,12 @@
+module.exports = (task) => (req, res, next) => {
+  req.user.can(task, req.params)
+    .then(allowed => {
+      if (allowed) {
+        return next();
+      }
+      const err = new Error('Not authorised');
+      err.status = 403;
+      return next(err);
+    })
+    .catch(next);
+};
```

Comment:
Can we rename this file `permissions.js`? It's not _really_ auth.

Reply:
why don't we just call all the files auth?